File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    num1 = [1,2,3,4,5]
    num2 = [2,1,2,3,2,4,2,5]
    num3 = [3,3,1,1,2,2,4,4,5,5]
    result = [0,0,0]

    check = 0
    check1 = 0
    check2 = 0
    check3 = 0

    while check < len(answers):
        if answers[check] == num1[check1]:
            result[0] += 1
        if answers[check] == num2[check2]:
            result[1] += 1
        if answers[check] == num3[check3]:
            result[2] += 1

        if check1+1 == len(num1):
            check1 = -1

        if check2+1 == len(num2):
            check2 = -1

        if check3+1 == len(num3):
            check3 = -1
        check += 1
        check1 += 1
        check2 += 1
        check3 += 1
    check = 0
    while check< len(result):
        if result[check] == max(result):
            answer.append(check+1)
        check += 1
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public List<Integer> solution(int[] answers) {
        List<Integer> answer = new ArrayList<>();
        
        int [] answerOne = { 1, 2, 3, 4, 5};
        int [] answerTwo = {2, 1, 2, 3, 2, 4, 2, 5};
        int [] answerThree = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        int personOne = 0;
        int personTwo = 0;
        int personThree = 0;
        int [] result = {0, 0, 0};
        
        
        for(int i =0; i<answers.length; i++){
            if(answerOne[personOne] == answers[i]){
                result[0] ++;
            }
            if(answerTwo[personTwo] == answers[i]){
                result[1] ++;
            }
            if(answerThree[personThree] == answers[i]){
                result[2] ++;
            }
            
            personOne ++;
            personTwo ++;
            personThree ++;
            
            if(personOne >= answerOne.length){
                personOne = 0;
            }
            if(personTwo >= answerTwo.length){
                personTwo = 0;
            }
            if(personThree >= answerThree.length){
                personThree = 0;
            }
            
        }
        
        for(int i =0; i<result.length; i++){
            if(result[i] == Arrays.stream(result).max().getAsInt()){
                answer.add(i+1);
            }
        }
        
        return answer;
    }
}